refactored documentation (#51)
oindrillac/ttmtool: README.md

[![TTM Inference](https://github.com/redhat-et/time-to-merge-tool/actions/workflows/inference.yaml/badge.svg)](https://github.com/redhat-et/time-to-merge-tool/blob/main/.github/workflows/inference.yaml)

# Github Action for Time to Merge Model

> This repository contains a GitHub Action to train the Github time to merge model. This model can be trained on any repository and be used to predict the time to merge of new pull requests. To learn more about this model, please see [here](https://github.com/aicoe-aiops/ocp-ci-analysis/tree/master/notebooks/time-to-merge-prediction).

Reviewer: ```suggestion
**Want an estimate on how long it will take to get your PR merged?**

The data science team within Red Hat's Emerging Technologies group wanted to know if we could leverage Github Actions to help us do some predictive analytics and measure the velocity of our projects. To that end we developed the "Time to Merge" (TTM) tool .    

This repository contains the TTM GitHub Action needed to train a custom model for individual projects and provide predictions in the form of PR comments. This model can be trained on any repository and can be used to predict the time to merge of new pull requests. To learn more about this approach, please see [here](https://github.com/aicoe-aiops/ocp-ci-analysis/tree/master/notebooks/time-to-merge-prediction).
```

> To use the Github Action for your own repository and train the model, you can follow these steps:

Reviewer: ```suggestion
To use the Github Action for your own repository and train a model, follow these steps:
```

### Pre-requisites:

> 1. **S3 bucket credentials**: You will need an S3 bucket to store the data and the model generated as a part of the training process. You can pass the S3 bucket credentials in 2 ways. You can either set them up as Github Action Secrets or pass them as a payload from your http request.

Reviewer: ```suggestion
1. **S3 bucket**: You will need an S3 bucket to store the data and the model generated as a part of the training process. You can pass the S3 bucket credentials in 2 ways. You can either set them up as Github Action Secrets or pass them as a payload from your http request.
```

## Step 1

Once you have the pre-requisites in place, add your S3 credentials to your repository action secrets (this is the recommended approach) if they are private and you dont want to pass them on through the http request .

To do that, go to your repository "Settings" -> "Security" -> "Secrets" -> "Actions" -> "New Repository Secret" and add secrets for `S3_BUCKET`, `S3_ENDPOINT_URL`, `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, your personal access token as your `GITHUB_TOKEN`, the prefix/folder where you want to store the data on the S3 bucket as the `CEPH_BUCKET_PREFIX`, and the GitHub repository that you want to train the model on and the organization it belongs to as `REPO` and `ORG`.

<img width="200" alt="image" src="https://user-images.githubusercontent.com/32435206/195929605-4518559e-7ffd-4b6d-a47f-e06fd1cdb4ac.png">

<img width="500" alt="image" src="https://user-images.githubusercontent.com/32435206/195929854-840a5784-a23a-4412-b23e-1b83c0160e75.png">

## Step 2

> To use this Github Action, you would need to first train your time to merge model and then use it on new pull requests on your repository. This can be done by adding 2 workflow files to your repository and running them in 2 modes.

Reviewer: ```suggestion
To use this Github Action for both training and inference on new PR's, you will need to add 2 workflow files to your repository.

One action controls the training workflow, and is triggered on-demand and as needed. The other action controls the inference workflow, and is triggered on each new PR submission. 

```

1. **Training Mode** :

For every new repository, you need to first train the model on the historical pull requests. 

> To do that, you will need to add a `train-ttm.yaml` file to the `.github/worklows/` folder on your repository like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/train-ttm.yaml). To run the action in training mode, specify the `MODE` as `1`. 

Reviewer: ```suggestion
To do that, you will need to add a `train-ttm.yaml` file to the `.github/worklows/` folder on your repository that looks like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/train-ttm.yaml). To run the action in training mode, make sure you specify the `MODE` as `1`. 
```

2. **Inference Mode** : 

> Similar to the `train-ttm.yaml` file, you can add another file called `predict-ttm.yaml` file to `.github/worklows/` folder in your repository like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/predict-ttm.yaml). This file has `MODE` as `0` which would enable just inference on the new incoming pull requests and add a comment on the pull request specifying the approximate time it will take to be merged.

Reviewer: ```suggestion
Similar to the `train-ttm.yaml` file, you need to add another file called `predict-ttm.yaml` to the `.github/worklows/` folder in your repository that looks like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/predict-ttm.yaml). This file has set `MODE` to `0` which will enable inference on all new incoming pull requests and add a comment on the pull request specifying the approximate time it will take to be merged.
```

## Alternate Approach

> You can also use train this model on your repository using an alternate approach without adding the workflow file to your repository. Here are the steps that you can follow:

Reviewer: ```suggestion
You can also use this tool on your repository with an alternate approach without adding the workflow file to your repository. Here are the steps that you can follow:
```
